File: src/main/scala/com/example/scala/Greeter.scala
package com.example.scala

import akka.actor.Actor
import akka.actor.actorRef2Scala

object Greeter {
  case object Greet
  case object Done
}

class Greeter extends Actor {
  def receive = {
    case Greeter.Greet =>
      println("Hello, world")
      sender ! Greeter.Done
  }
}

File: src/main/java/com/example/java/Greeter.java
package com.example.java;

import akka.actor.*;

public class Greeter extends UntypedActor {
    public enum Msg {
        GREET, DONE;
    }

    @Override
    public void onReceive( Object msg ) throws Exception {
        if ( msg == Msg.GREET ) {
            System.out.println( String.format( "Hello World from %s", getClass().getName() ) );
            getSender().tell( Msg.DONE, getSelf() );
        } else {
            unhandled( msg );
        }
    }
}
